lib/components/stand-form/form.tsx

```diff
@@ -1,138 +1,148 @@
 "use client";
 
-import {
-	Card,
-	CardContent,
-	CardHeader,
-	CardTitle,
-} from "@/lib/components/ui/card";
-import { Tabs, TabsTrigger } from "@/lib/components/ui/tabs";
-import { TabsList } from "@/lib/components/ui/tabs";
-import {
-	createContext,
-	Dispatch,
-	SetStateAction,
-	useEffect,
-	useState,
-} from "react";
-import {
-	AutoTab,
-	EndgameTab,
-	MiscTab,
-	tabMappings,
-	tabs,
-	TeleopTab,
-} from "./tabs";
-import { Button } from "@/lib/components/ui/button";
+import { KeyboardEvent, useEffect } from "react";
 import {
 	StandFormData,
-	StandFormValidationResult,
-	validate,
-	ValidationContext,
-} from "@/lib/components/stand-form/validate";
-import { TabsContentForceMount } from "./force-mount-tab";
-import { useFormState } from "react-dom";
-import { ServerActionResult } from "@/lib/actions/actions-utils";
-import { ValidatedLabel } from "../ui/label";
+	standFormSchema,
+} from "@/lib/components/stand-form/client-validate";
 import { Input } from "../ui/input";
+import { useFormContext } from "react-hook-form";
+import { AutoForm } from "../forms/autogenerate";
+import { Textarea } from "../ui/textarea";
+import { CounterInput } from "../forms/counter-input";
+import {
+	Select,
+	SelectContent,
+	SelectItem,
+	SelectTrigger,
+	SelectValue,
+} from "../ui/select";
+import { ServerActionResult } from "@/lib/actions/actions-utils";
+import { standFormAction } from "./server-validate";
 
-// TODO: Add form submission logic
+const trimInput = (e: KeyboardEvent<HTMLInputElement>, maxLength: number) => {
+	if (e.code === "Minus") {
+		e.preventDefault();
+	}
+
+	if (e.currentTarget.value.length >= maxLength) {
+		e.currentTarget.value = e.currentTarget.value.slice(0, maxLength - 1);
+	}
+};
 
 export function StandForm({
-	handleSubmit,
+	onSubmit,
 }: {
-	handleSubmit: (data: StandFormData) => Promise<ServerActionResult<unknown>>;
+	onSubmit: (data: StandFormData) => Promise<ServerActionResult<unknown>>;
 }) {
-	const [activeTab, setActiveTab] = useState("auto");
-	const [formState, formAction] = useFormState(
-		async (_: unknown, formData: FormData) => {
-			formData.append("scouter", "e22e61b8-bd4d-425d-8d6b-ba1b7e93c2e0");
-			formData.append("event_code", "2024test");
-
-			const result = validate(formData);
-
-			if (result.errors.length) {
-				return result;
-			}
-
-			await handleSubmit(result.data!);
-
-			return result;
-		},
-		{ errors: {}, data: null },
-		"/"
-	);
-
-	useEffect(() => {
-		const activeErrorTab = Object.keys(tabMappings).find((tabKey) =>
-			Object.keys(formState.errors).some((errorKey) =>
-				errorKey.includes(tabKey)
-			)
-		);
+	return (
+		<div className="flex justify-center">
+			<AutoForm
+				title="Stand Form"
+				data={standFormSchema}
+				ui={{
```

Comment:
Can we define the UI elsewhere?

Reply:
I think it'd make it a bit easier to look through the code if we had a config file that let us specify our fields. Happy to explain more on that.